sup3r/qa/stats.py

```diff
@@ -0,0 +1,494 @@
+            logger.info('Initializing qa output file: "{}"'.format(qa_fp))
+            with open(qa_fp, 'wb') as f:
+                pickle.dump(data, f)
+        else:
+            logger.info(f'{qa_fp} already exists.')
+
+    @classmethod
+    def get_node_cmd(cls, config):
+        """Get a CLI call to initialize Sup3rWindStats and execute the
+        Sup3rWindStats.run() method based on an input config
+
+        Parameters
+        ----------
+        config : dict
+            sup3r wind stats config with all necessary args and kwargs to
+            initialize Sup3rWindStats and execute Sup3rWindStats.run()
+        """
+        import_str = 'import time;\n'
+        import_str += 'from reV.pipeline.status import Status;\n'
+        import_str += 'from rex import init_logger;\n'
+        import_str += 'from sup3r.qa.stats import Sup3rWindStats;\n'
+
+        qa_init_str = get_fun_call_str(cls, config)
+
+        log_file = config.get('log_file', None)
+        log_level = config.get('log_level', 'INFO')
+
+        log_arg_str = (f'"sup3r", log_level="{log_level}"')
+        if log_file is not None:
+            log_arg_str += f', log_file="{log_file}"'
+
+        cmd = (f"python -c \'{import_str}\n"
+               "t0 = time.time();\n"
+               f"logger = init_logger({log_arg_str});\n"
+               f"qa = {qa_init_str};\n"
+               "qa.run();\n"
+               "t_elap = time.time() - t0;\n")
+
+        job_name = config.get('job_name', None)
+        if job_name is not None:
+            status_dir = config.get('status_dir', None)
+            status_file_arg_str = f'"{status_dir}", '
+            status_file_arg_str += f'module="{ModuleName.WIND_STATS}", '
+            status_file_arg_str += f'job_name="{job_name}", '
+            status_file_arg_str += 'attrs=job_attrs'
+
+            cmd += ('job_attrs = {};\n'.format(json.dumps(config)
+                                               .replace("null", "None")
+                                               .replace("false", "False")
+                                               .replace("true", "True")))
+            cmd += 'job_attrs.update({"job_status": "successful"});\n'
+            cmd += 'job_attrs.update({"time": t_elap});\n'
+            cmd += f'Status.make_job_file({status_file_arg_str})'
+
+        cmd += (";\'\n")
+
+        return cmd.replace('\\', '/')
+
+    @staticmethod
+    def get_wind_stats(u, v):
+        """Get stats for wind fields
+
+        Parameters
+        ----------
+        u: ndarray
+            Longitudinal velocity component
+            (lat, lon, temporal)
+        v : ndarray
+            Latitudinal velocity component
+            (lat, lon, temporal)
+
+        Returns
+        -------
+        stats : dict
+            Dictionary of stats for wind fields
+        """
+        v_dist = velocity_gradient_dist(u)
+        vort_dist = vorticity_dist(u, v)
+        rr_dist = ramp_rate_dist(u, v)
```

Comment:
Hey i actually have some questions about the ramp rate dist function. 

- if you want to keep this as u/v ramps, we should rename it to ws_ramp_rate_dist. 
- We should set the diff screening criteria to a kwarg (`diffs = diffs[(np.abs(diffs) < 10)]`)
- We should set the difference window to a kwarg (6 and 12 hour ramp rates are just as important for power systems work but may not be useful for wind-actuals validation)
- Why do you perform this RMSE operation: `diffs = diffs / np.sqrt(np.mean(diffs**2))`? This makes the output units not values of m/s right?
- I think we should be normalizing the counts output to `counts.sum()` not `counts.max()`. The former is more typical of PDF's isnt it?

Reply:
Yeah good points about kwargs. The RSME operation is used in the initial superres paper. Also good point about norming by the sum.

Reply:
That's fine if we want normalized values for the ramp rate / velocity gradient but we should not represent our outputs as windspeeds in m/s then and we should change the output docstring (the outputs are not delta windspeed or delta u but a normalized version of that). The normalization makes sense for problems where you're changing the spatiotemporal resolution but makes the outputs more difficult to interpret when you're benchmarking against a similar dataset with the same target resolution.

Reply:
Yeah good point. I'll update the doc string.